StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/MandateTextProvider.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  MandateTextProvider.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 10/3/24.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripeUICore
+
+protocol MandateTextProvider {
+    func mandate(for paymentMethodType: PaymentSheet.PaymentMethodType?, savedPaymentMethod: STPPaymentMethod?, bottomNoticeAttributedString: NSAttributedString?) -> NSAttributedString?
+}
+
+/// A class that can provide the attributed string for a given payment method type and configuration
+class FormMandateProvider: MandateTextProvider {
+    private let configuration: PaymentElementConfiguration
+    private let elementsSession: STPElementsSession
+    private let intent: Intent
+
+    init(configuration: PaymentElementConfiguration, elementsSession: STPElementsSession, intent: Intent) {
+        self.configuration = configuration
+        self.elementsSession = elementsSession
+        self.intent = intent
+    }
+
+    /// Builds the attributed string for a given payment method type
+    /// - Parameter paymentMethodType: The payment method type who's mandate should be constructed
+    /// - Parameter savedPaymentMethod: The currently selected saved payment method if any
+    /// - Parameter bottomNoticeAttributedString: The 
```

Comment:
nit this is missing

Reply:
"Passing this in just makes this method return it as long as `configuration` doesn't hide mandate text"